design/file-descriptors.md

```diff
@@ -0,0 +1,14 @@
+# File Descriptors
+File Descriptors ([witx type `fd`](https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#-fd)) are an abstract way of representing access to a resource. More often than not, they are processes (WIP), sockets (WIP), files, or even the console.
+
+# The Console
+When a program is initiated, `stdout`, `stdin`, and `stderr` (respectively `fd` ids `0`, `1`, and `2`) are opened. To print to these, simply call `fd_write` with the corresponding arguments. An simple "Hello World!" program written in `wat` is shown below:
```

Comment:
Shouldn't the order of file descriptors be `stdin`, `stdout`, `stderr`? That's the way I've seen it on every other OS that I can think of.

Reply:
Good catch -- yes, it should be `stdin`, `stdout`, `stderr`.

Reply:
Fixed!